appservice/src/createAppService/CustomLocationListStep.ts

```diff
@@ -0,0 +1,63 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtLocation, IAzureQuickPickItem, IAzureQuickPickOptions, LocationListStep, parseError } from 'vscode-azureextensionui';
+import { localize } from '../localize';
+import { createResourceGraphClient } from '../utils/azureClients';
+import { CustomLocation, IAppServiceWizardContext } from './IAppServiceWizardContext';
+
+export class CustomLocationListStep<T extends IAppServiceWizardContext> extends LocationListStep<T> {
+    public async prompt(wizardContext: T): Promise<void> {
+        const options: IAzureQuickPickOptions = { placeHolder: localize('selectLocation', 'Select a location for new resources.'), enableGrouping: true };
+        const result: AzExtLocation | CustomLocation = (await wizardContext.ui.showQuickPick(this.getCustomQuickPicks(wizardContext), options)).data;
+        if ('kubeEnvironment' in result) {
+            wizardContext.customLocation = result;
+            // For any resources other than the app, we still need a non-custom location, so we'll use the kubeEnvironment's location
+            await LocationListStep.setLocation(wizardContext, result.kubeEnvironment.location);
```

Comment:
Is there any negative impact of setting the location if this is app? Though the comment mentions it, there's no specific check to verify that this isn't an app.